metron-analytics/metron-profiler-client/README.md

```diff
@@ -91,37 +60,268 @@ want to change the global Client configuration so as not to disrupt the work of
+            of the same name. Default is the empty Map, meaning no overrides.
+
+e.g. To retrieve all the measurements written for 'profile' and 'entity' for the last hour 
+on the same weekday excluding weekends and US holidays across the last 14 days: 
+PROFILE_GET('profile', 'entity', PROFILE_WINDOW('1 hour window every 24 hours starting from 14 days ago including the current day of the week excluding weekends, holidays:us'))
+```
+
+Note that the `config_overrides` parameter operates exactly as the `config_overrides` argument in `PROFILE_GET`.
+The only available parameters for override are:
+* `profiler.client.period.duration`
+* `profiler.client.period.duration.units`
+
+##### The Profile Selector Language
+
+The domain specific language can be broken into a series of clauses, some optional
+* <span style="color:blue">Total Temporal Duration</span> - The total range of time in which windows may be specified
+* <span style="color:red">Temporal Window Width</span> - How large each temporal window
+* <span style="color:green">Skip distance</span> (optional)- How far to skip between when one window starts and when the next begins
+* <span style="color:purple">Inclusion/Exclusion specifiers</span> (optional) - The set of specifiers to further filter the window
+
+One *must* specify either a total temporal duration or a temporal window width.
+The remaining clauses are optional.
+During the course of the following discussion, we will color code the clauses in the examples.
+
+From a high level, the language fits the following three forms:
+
+* <span style="color:red">`time_interval WINDOW?`</span><span style="color:purple">`(INCLUDING specifier_list)? (EXCLUDING specifier_list)?`</span>
+* <span style="color:red">`time_interval WINDOW?`</span><span style="color:green">`EVERY time_interval`</span><span style="color:blue">`FROM time_interval (TO time_interval)?`</span><span style="color:purple">`(INCLUDING specifier_list)? (EXCLUDING specifier_list)?`</span>
+* <span style="color:blue">`FROM time_interval (TO time_interval)?`</span>
+
+with
+* `time_interval` representing a time amount followed by a unit (e.g. "1 hour")
+* `specifier_list` representing a comma separated list of inclusion or exclusion specifiers (e.g. "holidays:us, tuesdays")
+
+
+###### <span style="color:blue">Total Temporal Duration</span>
+
+Total temporal duration is specified by a phrase: `FROM time_interval AGO TO time_interval AGO`
+This indicates the beginning and ending of a time interval.
+* `FROM` - Can be the words "from" or "starting from"
+* `time_interval` - A time amount followed by a unit (e.g. 1 hour).  The unit may be "minute", "day", "hour" with any pluralization.
+* `TO` - Can be the words "until" or "to"
+* `AGO` - Optionally the word "ago"
+
+The `TO time_interval AGO` portion is optional.  If unspecified then it is expected that the time interval ends now.
```

Comment:
One thing to consider is that we seem to be assuming "processing time" with the grammar.   For example, in the testing notes that you provided, we are enriching the message with an expression like this.

```
STATS_MEAN(STATS_MERGE(PROFILE_GET('stat', 'global', PROFILE_WINDOW('5 minute window every 10 minutes starting from 2 minutes ago until 32 minutes ago excluding holidays:us'))))
```

When we grab the profile from "2 minutes ago" this will be 2 minutes ago from processing time (aka system time) rather than the event time of the message.

Is there any way to support event time now?  I'm not sure that this is needed right now, but thought we should have a discussion around this at the very least.